utils/cr_demo/demo_utils.py

```diff
@@ -0,0 +1,396 @@
+"""
+  (C) Copyright 2023 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+import subprocess
+import yaml
+
+
+# Storage-related methods
+def format_storage(host_list):
+    """Call dmg storage format.
+
+    Args:
+        host_list (str): List of hosts to format.
+    """
+    format_cmd = ["dmg", "storage", "format", "--host-list=" + host_list]
+    run_command(command=format_cmd)
+
+
+def storage_query_usage(host_list):
+    """Call dmg storage query usage.
+
+    Args:
+        host_list (str): List of hosts to query.
+    """
+    storage_query_cmd = ["dmg", "storage", "query", "usage", "--host-list=" + host_list]
+    run_command(command=storage_query_cmd)
+
+
+# Pool-related methods
+def create_pool(pool_size, pool_label, ranks=None, nsvc=None):
+    """Call dmg pool create.
+
+    Args:
+        pool_size (str): Pool size.
+        pool_label (str): Pool label.
+        ranks (str): Ranks to create pool. Defaults to None.
+        nsvc (str): Number of service replicas. Defaults to None.
+    """
+    create_pool_cmd = ["dmg", "pool", "create", pool_label, "--size=" + pool_size]
+    if ranks:
+        create_pool_cmd.append("--ranks=" + ranks)
+    if nsvc:
+        create_pool_cmd.append("--nsvc=" + nsvc)
+    run_command(command=create_pool_cmd)
+
+
+def list_pool(verbose=False, json=False, no_query=False):
+    """Call dmg pool list.
+
+    Args:
+        verbose (bool): Whether to use --verbose. Defaults to False.
+        json (bool): Whether to use --json. If used, verbose value would be irrelevant.
+            Defaults to False.
+        no_query (bool): Whether to use --no-query. Defaults to False.
+
+    Returns:
+        str: If --json is used, return stdout. Otherwise None.
+
+    """
+    if json:
+        list_pool_cmd = ["dmg", "--json", "pool", "list"]
+    else:
+        list_pool_cmd = ["dmg", "pool", "list"]
```

Comment:
Optional
`--json` can come after. E.g.
```suggestion
    list_pool_cmd = ["dmg", "pool", "list"]
    if json:
        list_pool_cmd.append("--json")
```